lib/thesis/uploaders/repo_uploader.ex

```diff
@@ -0,0 +1,23 @@
+defmodule Thesis.RepoUploader do
+  @moduledoc """
+  Handles file uploads
+  """
+
+  @behaviour Thesis.Uploader
+
+  import Thesis.Config
+  import Ecto.Query, only: [from: 2]
+  alias Thesis.File
+
+  def upload(file) do
+    changeset = File.changeset(%File{}, file)
+    do_upload(changeset)
+  end
+
+  defp do_upload(changeset) do
+    case repo.insert_or_update(changeset) do
+      {:ok, file} -> "/thesis/files/" <> file.slug
+      {:error, _} -> ""
```

Comment:
@jamonholmgren well, i changed this specific function, but on the upload_file in the controller, we are giving a response back to react and we change the state. I have logic that checks if the string has length.  
https://github.com/infinitered/thesis-phoenix/pull/79/files#diff-ba61dd9898e9cfe3b15d067e3df6f8c1R79